src/main/java/uk/gov/hmcts/cmc/claimstore/services/DocumentManagementService.java

```diff
@@ -0,0 +1,84 @@
+public class DocumentManagementService {
+
+    protected static final String APPLICATION_PDF = "application/pdf";
+    protected static final String PDF_EXTENSION = ".pdf";
+    protected static final String FILES_NAME = "files";
+    private final DocumentMetadataDownloadClientApi documentMetadataDownloadApi;
+    private final DocumentDownloadClientApi documentDownloadClientApi;
+    private final DocumentUploadClientApi documentUploadClientApi;
+    private final ClaimRepository claimRepository;
+
+    @Autowired
+    public DocumentManagementService(
+        final DocumentMetadataDownloadClientApi documentMetadataDownloadApi,
+        final DocumentDownloadClientApi documentDownloadClientApi,
+        final DocumentUploadClientApi documentUploadClientApi,
+        final ClaimRepository claimRepository
+    ) {
+        this.documentMetadataDownloadApi = documentMetadataDownloadApi;
+        this.documentDownloadClientApi = documentDownloadClientApi;
+        this.documentUploadClientApi = documentUploadClientApi;
+        this.claimRepository = claimRepository;
+    }
+
+    public void storeClaimN1Form(final String authorisation, final Claim claim, final byte[] n1FormPdf) {
+        final Document document = uploadDocument(authorisation, claim, n1FormPdf);
+
+        claimRepository.linkDocumentManagement(claim.getId(), URI.create(document.links.self.href).getPath());
+    }
+
+    public byte[] getClaimN1Form(final String authorisation, final Claim claim, final byte[] n1FormPdf) {
+        if (StringUtils.isBlank(claim.getSealedClaimDocumentManagementSelfUri())) {
+            storeClaimN1Form(authorisation, claim, n1FormPdf);
+            return n1FormPdf;
+        }
+
+        final Document documentMetadata = documentMetadataDownloadApi.getDocumentMetadata(authorisation,
+            claim.getSealedClaimDocumentManagementSelfUri());
+
+        final ResponseEntity<Resource> responseEntity = documentDownloadClientApi.downloadBinary(authorisation,
+            URI.create(documentMetadata.links.binary.href).getPath());
+
+        if (responseEntity.getStatusCode().equals(HttpStatus.NOT_FOUND)) {
+            storeClaimN1Form(authorisation, claim, n1FormPdf);
+            return n1FormPdf;
+        } else {
+            return ((ByteArrayResource) (responseEntity.getBody())).getByteArray();
+        }
+    }
+
+    private Document uploadDocument(final String authorisation, final Claim claim, final byte[] n1FormPdf) {
+        final String originalFileName = claim.getReferenceNumber() + PDF_EXTENSION;
+        final MultipartFile file = new InMemoryMultipartFile(FILES_NAME, originalFileName, APPLICATION_PDF, n1FormPdf);
+        final List<MultipartFile> files = Collections.singletonList(file);
+        final UploadResponse response = documentUploadClientApi.upload(authorisation, files);
+
+        return response.getEmbedded().getDocuments().stream()
+            .findFirst()
+            .orElseThrow(IllegalArgumentException::new);
```

Comment:
Why is that `IllegalArgumentException`? The logic seems a bit unexpected to me. What is the scenario documents stream is empty? What would be the status code in that scenario?